src/tsdl.ml

```diff
@@ -2279,8 +2279,18 @@ let gl_get_drawable_size win =
   gl_get_drawable_size win w h;
   (!@ w, !@ h)
 
+let nat_to_maybe_error =
+  let read = function
+       | n when n < 0 ->
+          (match error () with
+          | Error (`Msg "") -> Ok n
+          | err -> err)
+      | n -> Ok n in
+  view ~read ~write:write_never int
+
 let gl_get_swap_interval =
-  foreign "SDL_GL_GetSwapInterval" (void @-> returning nat_to_ok)
+  clear_error ();
```

Comment:
Why do you clear the error here ?

Reply:
The documentation and implementation of SDL is not very clear on what happens when a negative value is returned from this function.
Functions don't clear the error on success (this is documented here: https://wiki.libsdl.org/SDL2/SDL_GetError),
and since -1 is now a valid return value there isn't anything to distinguish that from an error :(
Here is an example from the 'x11' driver where it can return a negative value without setting or clearing the error: https://github.com/libsdl-org/SDL/blob/SDL2/src/video/x11/SDL_x11opengl.c#L939-L941

SDL3 will have a better API (0/-1 return value, and changing an 'int* ' argument), but with SDL2 we're stuck with what we have.
In fact I'm wondering whether we'd be better off just always returning Ok here, since this function clearly cannot fail in a useful way, and according to https://wiki.libsdl.org/SDL2/SDL_GetError again even successful returns may have temporarily set an error, so my approach above could still conclude there is an error when there isn't one.

(I don't want to change the API of 'gl_get_swap_interval' so this'd be just an unconditional 'turn the int into an Ok').
What do you think?

Reply:
I've pushed some fixups implementing the above idea, let me know if you like it and I can squash the fixups then